[FEAT] 장르 전체 조회 기능 및 테스트 코드 추가

twotwobread opened the pull request:
## PR Description 🗒️

## 추가/변경 사항 및 설명 📝
- 회의에서 나왔던 장르 타입들 모두 반영.
- 장르 enum 값과 장르명을 표현하기 위한 GenreObjectResponseDto 추가.
- 모든 장르 리스트를 반환하기 위한 GenreAllResponseDto 추가.
- 장르 전체 조회 기능을 수행하기 위한 컨트롤러 생성 -> POJO에 대한 조회 기능이기에 컨트롤러 단에서 로직 처리.
- 추가되는 코드들에 대한 테스트 추가.

## To Reviewers 🙏
- 바뀐 파일이 쓸데없는 것들이 많은데 이 부분 죄송함다,,, 분류해서 코드 리뷰 잘 부탁드립니다!! ㅠㅠ
## Issue close ✂️
#38

twotwobread committed:
[DOCS] PR template 추가
- .github 하위에 pr template 추가.

close #1이수영_#1

twotwobread committed:
[CHORE] jacoco 설정을 위한 build.gradle 파일 수정.
- jacoco plugins 추가
- jacocoTestReport task 추가
- jacocoTestCoverageVerification task 추가
- test task 실행되고 추가한 두 task가 실행될 수 있게 finalizedBy 추가
close #이수영_#3

twotwobread committed:
[CHORE] 코드 커버리지에서 lombok 어노테이션을 제외하기 위한 설정파일 추가

close #이수영_#3

twotwobread committed:
[DEL] 해당 PR에 필요없는 부분 삭제

twotwobread committed:
[CHORE] 필요없는 주석 삭제

twotwobread committed:
Merge branch 'develop' into lsy

twotwobread committed:
Merge branch 'develop' into suyoung

twotwobread committed:
Merge branch 'develop' into suyoung

twotwobread committed:
[TEST] : 필요없는 테스트 삭제.

twotwobread committed:
[FEAT] Member 엔티티 추가.
- 대체 키를 이용하도록 구성.
- 랭킹, 포인트, 승수에 대한 부분 숫자로 표현
- 프로필 이미지를 위한 이미지 필드 추가.
- 멤버가 참여한 대결에 대한 정보를 위해 battles 추가.
- 멤버가 공유한 게시글에 대한 정보를 위해 posts 추가.

twotwobread committed:
[FEAT] Image 엔티티 추가.
- 대체 키 이용.
- 이미지 경로 String으로  추가.
- 해당 이미지를 프로필로 가진 멤버에 대한 부분 일대일 관계 설정.
- 멤버측에서 더 많은 작업이 발생할 것 같아서 외래키 주인을 멤버로 설정.

twotwobread committed:
[FEAT] Music 엔티티 추가.
- 대체키 이용.
- 가수 이름을 저장하는 필드 추가.
- 음악 제목을 저장하는 필드 추가.
- 장르를 enum 형식으로 저장.
- 음악 재생과 관련된 url을 저장.
- 해당 음악을 공유하는 게시글에 대한 정보를 일대다 관계로 설정.

- Music 엔티티가 있어야할까란 생각도 듬. 그 이유는 어짜피 Music을 이용한 기능이 없고 이렇게 되면 음악 검색 api를 이용하기 전에 존재하는 music인지부터 확인하는 작업이 필요해질것 같음.

twotwobread committed:
[FEAT] Post 엔티티 추가.
- 대체 키 이용.
- 대결 가능 여부에 대한 필드 추가.
- 좋아요 갯수를 숫자로 필드 추가.
- music에 대한 부분 다대일 관계로 설정.
- post에서 배틀이 필요한 이유는 배틀 쪽에서 Post의 정보가 필요해서 넣었음.

twotwobread committed:
[FEAT] Battle 엔티티 추가.
- 대체 키 이용
- 대결 신청을 받은 게시글의 득표 수에 대한 필드 추가.
- 대결을 신청한 게시글에 대한 득표 수 필드 추가.
- member가 필요한 이뉴는 member 측에서 자신이 참여한 battle의 정보를 가져오기 위함이다.
- battle에서는 두 개의 post 정보가 필요한데 각각 다른 필드로 구성되어도 될까?
  - 다르게 생각했던건 post를 List화 시킬 생각을 했는데 이것도 너무 이상하다.

twotwobread committed:
Merge pull request #31 from prgrms-web-devcourse/suyoung_#30

[FEAT] Entity 초기 설계

twotwobread committed:
[FIX] Battle 엔티티 수정.
- Post와 득표 수에 대한 필드를 임베디드 타입으로 설정.
- Member 필드 제거.

twotwobread committed:
[ADD] BaseEntity 추가.
- createdAt, updatedAt을 가진 BaseEntity를 이용하여 모든 엔티티에 생성, 수정 시간 추가.

twotwobread committed:
[FIX] Member 엔티티 수정.
- Image 엔티티 제거 -> Member 엔티티에 imageUrl로 추가.
- 필드명 일부 수정.
- 남은 대결권 필드 추가 -> remainingChallenges;
- 소셜 로그인과 관련된 필드 값 추가 -> refreshToken, social
- likes 리스트 필드 추가.

twotwobread committed:
[FIX] Music 관련 코드 수정.
- Music 엔티티 -> 임베디드 타입 클래스로 수정.
- post 관련 연관관계 삭제.

twotwobread committed:
[FIX] Post 엔티티 수정
- 필드 값 일부 수정.
- likes 리스트 필드 추가.

twotwobread committed:
[ADD] Like 엔티티 추가

twotwobread committed:
[ADD] Vote 엔티티 추가.

twotwobread committed:
[FIX] Battle 엔티티 수정.
- AttributeOverride를 이용한 컬럼명 오버라이드.
- AssociationOverride를 이용한 fk 오버라이드.

twotwobread committed:
[FIX] Like 엔티티 수정
- fk명 수정.

twotwobread committed:
[FIX] Member 엔티티 수정.
- 필드명 수정.
- fk명 수정.

twotwobread committed:
[FIX] MemberScore 클래스 수정.
- 필드명 수정.

twotwobread committed:
[FIX] Music 엔티티 수정.
- 필드명 수정.

twotwobread committed:
[FIX] Post 엔티티 수정.
- fk명 수정.
- 필드명 수정.

twotwobread committed:
[FIX] Vote 엔티티 수정.
- fk명 수정.

twotwobread committed:
[CHORE] guava 의존성 추가.

twotwobread committed:
[ADD] Battle 및 BattleInfo에 대한 validation 로직 추가.
- Bean Validation 추가.
- Guava를 통한 application validation 추가.
- 검증 로직
  - 득표 수가 음수가 되면 안됨.

twotwobread committed:
[ADD] Post, Music 및 Genre에 대한 validation 추가.
- Bean Validation 추가.
- Guava를 이용한 application validation 추가.
- 검증 로직
  - Post
    - 좋아요 개수가 음수이면 안됨.
  - Music
    - 음악 클래스는 모든 값이 무조건 필요하기 때문에 Builder를 이용하지 않았음. 근데 JPA 자체에서 기본 생성자를 이용하기에 어노테이션을 달아줌.
    - 음악 고유 번호가 Null이거나 공백으로 이루어지면 안됨.
    - 앨범 커버 이미지 URL이 Null or 공백이면 안되고 최대 2000자 까지
    - 가수 이름이 Null이거나 공백으로 이루어지면 안됨.
    - 음악 제목이 Null이거나 공백으로 이루어지면 안됨.
    - 장르가 Null이면 안됨.
    - 음악 URL이 Null or 공백이면 안되고 최대 2000자 까지

twotwobread committed:
[ADD] Member, MemberScore 및 SocialInfo, Social validation 추가.
- Bean Validation 추가.
- Guava를 이용한 application validation 추가.
- 검증 로직
  - Member
    - 프로필 이미지 URL이 Null or 공백이면 안되고 최대 2000자 까지
    - 닉네임이 Null or 공백이면 안되고 최소 1자부터 24자 까지
    - 대결권 개수가 음수일 수 없음.
    - refresh token이 Null or 공백이면 안됨.
    - 여기도 무조건 필요한 값이 있어서 Builder 말고 생성자를 이용하는게 더 좋아보이긴 함.
  - MemberScore
    - 기본 생성자를 이용하면 0으로 초기화될 것이고 ranking이 들어올수도 있고 아닐수도 있으니까 setter를 이용함.
    - 모든 필드값이 음수만 아니면 됨.
  - SocialInfo
    - 소셜 로그인 타입이 존재해야함.
    - 소셜 ID가 Null이거나 공백이면 안됨.

thguss committed:
[FIX] 최소 길이 제한 기준 변경

thguss committed:
[FIX] 회원 스코어 관련 업데이트 메소드 로직 추가 및 반영

thguss committed:
[FIX] String to Enum 타입으로 변경

thguss committed:
[DEL] 미사용 메소드 삭제

thguss committed:
[FIX] Music id 필드명 수정

thguss committed:
[FIX] 접근 지정자 변경

thguss committed:
[TEST] h2 테스트

thguss committed:
[FIX] 테이블명 변경 (예약어 이슈)

thguss committed:
Merge pull request #49 from prgrms-web-devcourse/feature_#48

[CHORE] Entity 설계 의논 및 반영

thguss committed:
[FIX] 그래들 버전 업그레이드

thguss committed:
[FIX] jacoco 활성화 off

thguss committed:
[FIX] jacoco 활성화 on

thguss committed:
[FIX] jacoco 활성화 off

thguss committed:
[ADD] copySwaggerUI 의존성 추가

twotwobread committed:
[ADD] Music 검증 로직 추가.
- 장르가 null인 경우에 대한 검증 로직 추가.

twotwobread committed:
[ADD] Post 생성자에서 받는 음악 고유 번호에 대한 변수명 수정.

twotwobread committed:
[TEST] PostValidationTest 추가.
- TestConstant로 재사용할 수 있을 것 같은 200자를 넘는 문자열 빼내기.
- Post에서 application level에서 검증하는 로직 추가.
- Music에 대한 application level에서의 검증 로직 추가.
- Like는 validation할 부분이 존재하지 않아서 테스트 코드 추가하지 않음.

twotwobread committed:
[FIX] 반복문을 활용하여 2000자보다 더 긴 문자열을 만드는 로직 구성.

twotwobread committed:
[FIX] @Lob 어노테이션 다시 붙이기

twotwobread committed:
[CHORE] line separators 변경 및 PostValidationTest를 PostTest로 이름 변경.

twotwobread committed:
[FIX] 생성 검증 expression, 메시지 수정.
- Music에 생성자 검증 로직에서 메시지 수정.
- SocialInfo 생성자 검증 로직에서 expression 수정.

twotwobread committed:
[TEST] Like 테스트 추가.
- Like 엔티티 생성자 검증 로직 추가.
- 검증 로직에 따른 validation 테스트 추가.

twotwobread committed:
Merge branch 'develop' into suyoung

# Conflicts:
#	build.gradle
#	src/main/java/com/example/demo/model/battle/BattleInfo.java
#	src/main/java/com/example/demo/model/member/Member.java
#	src/main/java/com/example/demo/model/member/MemberScore.java
#	src/main/java/com/example/demo/model/member/Social.java
#	src/main/java/com/example/demo/model/member/SocialInfo.java
#	src/main/java/com/example/demo/model/post/Genre.java
#	src/main/java/com/example/demo/model/post/Like.java
#	src/main/java/com/example/demo/model/post/Music.java
#	src/main/java/com/example/demo/model/post/Post.java

twotwobread committed:
[CHORE] lombok test에서도 사용할 수 있게 의존성 추가.

twotwobread committed:
[ADD] 장르 타입 추가.
- 회의에서 결정된 장르 모두 추가.

twotwobread committed:
[FIX] 장르에서 없어진 댄스를 이용하는 코드 수정.

twotwobread committed:
[ADD] 장르 관련 Dto 추가
- 장르의 value, name 정보를 내려주기 위한 GenreObjectResponseDto 추가.
- 모든 장르 리스트를 내려주기 위한 GenreAllResponseDto 추가.

twotwobread committed:
[TEST] 장르 DTO 관련 테스트 코드 추가.
- Dto 내부의 toEntity 메서드에 대한 테스트 추가.

twotwobread committed:
[FEAT] 장르 전체 조회 기능을 위한 컨트롤러 추가.
- 모든 장르를 내려주기 위해서 컨트롤러 추가.
- 장르는 enum으로 POJO이기에 따로 다른 레이어를 만들지 않고 컨트롤러에서 바로 처리하도록 구현.

twotwobread committed:
[TEST] 장르 컨트롤러에 대한 통합 테스트 추가.
- TestRestTemplate를 이용하여 Rest API에 대한 통합 테스트 작성.
- TestRestTemplate에 관한 자료 : https://easybrother0103.tistory.com/64

twotwobread committed:
Merge branch 'suyoung' into suyoung_#38

twotwobread committed:
Merge pull request #62 from prgrms-web-devcourse/suyoung_#38

[FEAT] 장르 전체 조회

twotwobread committed:
[CHORE] check style에 걸리는 부분 수정.

twotwobread committed:
Merge branch 'suyoung' of https://github.com/prgrms-web-devcourse/Team-06-Final-BE into suyoung

twotwobread committed:
[CHORE] check style에 걸리는 부분 수정.

twotwobread committed:
[FIX] 장르 전체 조회에 대한 url path 수정

twotwobread committed:
[CHORE] jacoco 코드 커버리지 설정 추가.

twotwobread committed:
Merge pull request #65 from prgrms-web-devcourse/suyoung_#64

[CHORE] jacoco 코드 커버리지 설정 추가.

park-se-jun commented on src/main/java/com/example/demo/model/post/Genre.java:
```diff
@@ -4,7 +4,19 @@
 @Getter
 public enum Genre {
-	DANCE("댄스");
+	HIPHOP_RAP("힙합/랩"),
```
굿굿

park-se-jun commented on src/test/java/com/example/demo/model/post/LikeTest.java:
```diff
@@ -49,7 +49,7 @@ private Post createPost() {
 		String singer = "뉴진스";
 		String title = "Ditto";
-		Genre genre = Genre.DANCE;
+		Genre genre = Genre.K_POP;
```
이건 kpop으로 햇군요👍️

park-se-jun commented on src/test/java/com/example/demo/model/post/PostTest.java:
```diff
@@ -14,7 +14,7 @@ public class PostTest {
 	private static final String singer = "뉴진스";
 	private static final String title = "Ditto";
-	private static final Genre genre = Genre.DANCE;
+	private static final Genre genre = Genre.ETC;
```
이건 etc로 햇군요👍️

twotwobread replied:
ㅋㅋㅋㅋ 그냥 아무 값이나 넣었습니다,,,

park-se-jun commented on src/main/java/com/example/demo/dto/genre/GenreAllResponseDto.java:
```diff
@@ -0,0 +1,19 @@
+public record GenreAllResponseDto(
+	List<GenreObjectResponseDto> genres
+) {
+	public static GenreAllResponseDto toEntity(Genre... genres) {
```
장르 리스트를 안밭고 이런식으로 받은 이유가 있나요?

twotwobread replied:
Genre.values() 메소드를 이용하게 되면 배열 형태로 나오기 때문에 이렇게 이용하는게 더 좋지 않을까 생각했는데 어떻게 생각하시나요?? 리스트 형태로 받게 되면 이제 배열을 리스트로 바꾸는 로직이 필요해서 이렇게 한번 써봤습니다!!

park-se-jun commented on src/main/java/com/example/demo/dto/genre/GenreObjectResponseDto.java:
```diff
@@ -0,0 +1,12 @@
+
+import com.example.demo.model.post.Genre;
+
+public record GenreObjectResponseDto(
```
GenreVo같은건어때요? 좀 별론가? 근데 이대로 해도 괜찮긴 할듯

twotwobread replied:
네이밍이 어렵더라구요,,, Vo 좋은거 같습니다!!

park-se-jun commented on src/test/java/com/example/demo/dto/genre/GenreAllResponseDtoTest.java:
```diff
@@ -0,0 +1,20 @@
+public class GenreAllResponseDtoTest {
+
+	@Test
+	public void 성공_모든_장르를_Dto로_바꿀수_있다() {
```
dto 테스트를 하셨네요. `모든 장르를` 보다 `장르 리스트를` 로 하는거 어떠신지요

twotwobread replied:
좋은거 같습니다!

park-se-jun commented on src/test/java/com/example/demo/dto/genre/GenreObjectResponseDtoTest.java:
```diff
@@ -0,0 +1,22 @@
+public class GenreObjectResponseDtoTest {
+
+	@Test
+	public void 성공_장르를_Dto로_변환할_수_있다() {
```
👍️

park-se-jun commented on src/main/java/com/example/demo/controller/genre/GenreController.java:
```diff
@@ -0,0 +1,23 @@
+public class GenreController {
+
+	@GetMapping("/genres")
+	public ResponseEntity<ApiResponse> getAllGenres() {
```
로직 없이 POJO 만 다루면 컨트롤러에서 바로 처리해도 되나 보군요

twotwobread replied:
그건 사실 해도 된다 안된다는 모르겠는데 이전에 듣기로는 컨트롤러단에서 레포지토리를 이용해도 상관없다는 말을 들었어서 그럼 최대한 단순하게 처리하는게 좋지 않을까라는 생각에 저는 이렇게 처리해봤습니다. 세준님의 생각은 어떠신가요??

park-se-jun commented on src/test/java/com/example/demo/controller/genre/GenreControllerIntegrationTest.java:
```diff
@@ -0,0 +1,42 @@
+	private int port;
+
+	@Test
+	public void 성공_모든_장르_정보를_조회할_수_있다() {
```
rest docs로 부탁 드려용 사용법이 약간 달라요

twotwobread replied:
넵 추가해서 다시 코드리뷰 요청드리겠습니다!!

twotwobread replied:
아 이 부분은 통합 테스트로 짠거라서 RestDocs를 이용한 MockMvc 테스트도 추가하겠습니다!!

twotwobread committed:
[FEAT] 유저 상세 정보 조회 기능 추가.
- member repository 추가.
- member controller 추가.

twotwobread committed:
[ADD] Member 상세 정보 응답 Dto 추가.

thguss commented on src/main/java/com/example/demo/dto/genre/GenreObjectResponseDto.java:
```diff
@@ -0,0 +1,12 @@
+	String genreName
+) {
+	public static GenreObjectResponseDto toObject(Genre genre) {
+		return new GenreObjectResponseDto(genre.name(), genre.getName());
```
오호 name()이 DANCE("댄스")라는 값이 있으면 `댄스`를 반환하는 것인 줄 알았는데, `DACNE`를 반환하나요?! 몰랐는데 알아갑니다 ㅇ0ㅇ !!

thguss commented on src/test/java/com/example/demo/controller/genre/GenreControllerIntegrationTest.java:
```diff
@@ -0,0 +1,42 @@
+		// then
+		assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+		assertThat(response.getBody().success()).isTrue();
+		assertThat(response.getBody()).isNotNull();
```
장르 컨트롤러 테스트는 Rest Docs 테스트를 하지 않아도 되나요?-?

twotwobread replied:
넵 추가하고 다시 코드리뷰 요청하겠습니다!!

thguss reviewed (commented):
수고하셨습니다!!

thguss commented:
파일 3개 정도가 충돌이 나는 것 같은데, 충돌 해결하고 로컬에서 빌드 돌려서 확인해주시면 좋을 것 같아요!

park-se-jun commented on src/main/java/com/example/demo/model/battle/BattleInfo.java:
```diff
@@ -17,7 +17,7 @@
 @Embeddable
 @NoArgsConstructor(access = AccessLevel.PROTECTED)
-public class BattleInfo {
+class BattleInfo {
```
public을 지워서 Battle 밖에서는 접근 못하게 하기 👍️

park-se-jun reviewed (approved):
고생하셨습니다

twotwobread committed:
[FIX] 코드 리뷰 반영 및 메소드명 변경
- dto 이름 수정.
- 테스트 메소드 이름 수정.
- dto 메소드명 수정.
- url path 수정.

twotwobread committed:
[TEST] 장르 전체 조회 기능에 대한 RestDocs 테스트 추가.

twotwobread committed:
Merge branch 'suyoung_#38' into suyoung

# Conflicts:
#	src/main/java/com/example/demo/controller/genre/GenreController.java

twotwobread committed:
Merge remote-tracking branch 'origin/develop' into suyoung

# Conflicts:
#	src/main/java/com/example/demo/model/member/SocialInfo.java
#	src/test/java/com/example/demo/model/post/LikeTest.java
#	src/test/java/com/example/demo/model/post/PostTest.java

twotwobread committed:
[CHORE] builde.gradle 충돌 해결

twotwobread committed:
[FIX] ExceptionMessage, MemberController 수정.
- ExceptionMessage Getter 추가.
- MemberController 공백 추가.

twotwobread committed:
[CHORE] 코드 커버리지 측정 기준(단위)를 클래스에서 패키지로 수정.
[CHORE] line seperator 변경

twotwobread committed:
[CHORE] @ParameterizedTest를 위한 설정 수정.

twotwobread committed:
Merge pull request #74 from prgrms-web-devcourse/suyoung_#73

[FIX] jacoco 설정파일 수정.

twotwobread committed:
[FEAT] 멤버가 공유한 모든 게시글을 가져오는 기능.
- Authentication에서 멤버의 ID를 가져오는 메소드 생성.
- Authentication을 Service 계층으로 보내는 것도 아닌 것 같고 Service 단에서 해줄 마땅한 로직이 없다고 생각되어서 컨트롤러 단에서 레포지토리를 이용.
- 레포지토리에서 유저 정보를 가져와 이를 DTO로 변환 후 반환.

twotwobread committed:
[FEAT] 멤버 레포지토리 추가.
- 멤버 레포지토리 추가.
- fetch join을 이용해서 가지고 있는 게시글을 한번에 들고 올 수 있게 구현.
  - 그 이유는 DTO로 변환 시 모든 게시글에 멤버에 nickname이 필요 그래서 DTO로 변환하면서 멤버의 닉네임을 넣어줘야 하여 fetch join을 사용하는 것이 더 효율적일 것이라 생각.

twotwobread committed:
[ADD] 음악 정보를 담기 위한 DTO 추가.

twotwobread committed:
[ADD] 멤버가 공유한 게시글 정보를 담기 위한 DTO 추가.

twotwobread committed:
[ADD] 멤버가 공유한 모든 게시글들을 담기 위한 DTO 추가.
- fetch 조인을 이용해서 여기서 posts를 가져올 때 쿼리가 날라가지 않고 이전 방식에선 MemberPostVoResponseDto에서도 멤버의 이름이 필요해 또 쿼리가 날라가야 한다고 생각하는데 외부에서 멤버 이름을 주입하는 방식으로 수정.

twotwobread committed:
[CHORE] 뮤직 접근 지정자 public으로 수정.
- Dto로 변환하기 위함.

twotwobread committed:
[CHORE] 필요없는 파라미터 삭제.

twotwobread committed:
[CHORE] Member의 필요없는 파라미터 제외 및 MemberScore 값을 얻기 위한getter 추가.

twotwobread committed:
[TEST] 유저 상세정보 조회에 관한 RestDocs 테스트 추가.

twotwobread committed:
[TEST] 멤버 컨트롤러 유닛 테스트 추가

twotwobread committed:
[TEST] 멤버 테스트를 위한 유틸 클래스 추가.

twotwobread committed:
[TEST] MockMvcRestDocumentationWrapper를 이용하게 RestDocs 테스트 수정.

twotwobread committed:
Merge branch 'suyoung' of https://github.com/prgrms-web-devcourse/Team-06-Final-BE into suyoung

twotwobread committed:
[TEST] MockMvcRestDocumentationWrapper로 document 이용하게 수정.

twotwobread committed:
Merge branch 'suyoung' into suyoung_#39

twotwobread committed:
Merge pull request #76 from prgrms-web-devcourse/suyoung_#39

[FEAT] 유저 세부정보 조회 기능 및 테스트 코드 추가.

twotwobread committed:
[FIX] Member 엔티티 업데이트 메소드 추가 및 남은 대결권 초기화 구문 수정.
- default로 설정되어있는 MemberScore의 값을 수정하기 위해 update 메소드 추가.
- 초기 멤버가 생성될 때 남은 대결권을 5개로 설정.

twotwobread committed:
[TEST] Member 테스트 수정.
- 내부 Member 생성자에 할당하지 않아도 되는 파라미터들 제거에 따른 테스트 수정.
- updateMemberSocre 메서드를 추가함에 따라 테스트 로직 수정.

thguss committed:
[FIX] 체크 스타일 반영

twotwobread committed:
[CHORE] line seperator LF로 수정.

twotwobread committed:
[TEST] 유저가 공유한 모든 게시글을 조회하는 기능에 대한 단위 테스트 구현.

twotwobread committed:
[FIX] 멤버 컨트롤러 로직 수정 -> 계층 추가.
- 멤버 서비스 추가 -> 서비스 계층없이 로직을 구성했을 때 테스트에서 더 힘들어졌음 -> 결국 로직이 잘못되었다라고 생각이 들었고 계층을 추가하여 극복!

twotwobread committed:
Merge branch 'suyoung' into suyoung_#40

# Conflicts:
#	build.gradle
#	src/main/java/com/example/demo/controller/member/MemberController.java
#	src/main/java/com/example/demo/model/member/Member.java

twotwobread committed:
[CHORE] checkstyle 맞추기 위한 수정.

twotwobread committed:
[FIX] 기존 유저 상세 정보 조회 로직 수정.

twotwobread committed:
[TEST] 유저 상세정보 조회 관련 테스트 코드 추가.
- 유저 상세정보 기능을 위한 단위 테스트 (MemberDetailsTest) 추가.
- 유저 상세정보 기능의 문서화를 위한 RestDocs 테스트 추가.

twotwobread committed:
[TEST] 유저가 공유한 게시글 조회 RestDocs에 예외 사항 추가.

twotwobread committed:
[ADD] PrincipalService 추가.
- Principal을 이용하여 Member의 정보를 얻기 위한 서비스 계층.

twotwobread committed:
Merge branch 'suyoung' into suyoung_#40

twotwobread committed:
Merge pull request #75 from prgrms-web-devcourse/suyoung_#40

[FEAT] 유저음악 공유 게시글 리스트 조회 기능 코드 추가.

twotwobread committed:
[TEST] 모든 장르를 조회하는 기능에 대한 테스트 스위트명 수정.

thguss reviewed (approved):
수고하셨습니다~!
